package/src/handler.test.ts

```diff
@@ -1,103 +1,76 @@
 import * as assert from 'assert'
-import { Handler } from './handler'
-import { Result } from './api'
-import { Position } from 'sourcegraph'
-
-interface SearchTest {
-    crossRepo?: boolean
-    doc: {
-        uri: string
-        text: string
-    }
-    expSearchQueries: string[]
-}
+import { referencesQueries, definitionQueries } from './handler'
+import { TextDocument } from 'sourcegraph'
 
 describe('search requests', () => {
     it('makes correct search requests for goto definition', async () => {
-        const tests: SearchTest[] = [
-            {
-                crossRepo: undefined,
-                doc: {
-                    uri: 'git://github.com/foo/bar?rev#file.c',
-                    text: 'token',
-                },
-                expSearchQueries: [
-                    '\\btoken\\b case:yes file:.(h|c|hpp|cpp|m|cc)$ type:file',
-                ],
-            },
-            {
-                crossRepo: true,
-                doc: {
-                    uri: 'git://github.com/foo/bar?rev#file.c',
-                    text: 'token',
-                },
-                expSearchQueries: [
-                    '\\btoken\\b case:yes file:.(h|c|hpp|cpp|m|cc)$ type:symbol',
-                    '\\btoken\\b case:yes file:.(h|c|hpp|cpp|m|cc)$ type:file',
-                ],
-            },
+        interface DefinitionTest {
+            doc: TextDocument
+            definitionPatterns?: string[]
+            expectedSearchQueries: string[]
+        }
+        const tests: DefinitionTest[] = [
             {
-                crossRepo: false,
                 doc: {
-                    uri: 'git://github.com/foo/bar?rev#file.c',
+                    uri: 'git://github.com/foo/bar?rev#file.cpp',
+                    languageId: 'cpp',
                     text: 'token',
                 },
-                expSearchQueries: [
-                    '\\btoken\\b case:yes file:.(h|c|hpp|cpp|m|cc)$ type:symbol repo:^github.com/foo/bar$@rev',
-                    '\\btoken\\b case:yes file:.(h|c|hpp|cpp|m|cc)$ type:file',
+                definitionPatterns: ['const\\s%s\\s='],
+                expectedSearchQueries: [
+                    // current file
+                    'const\\stoken\\s= case:yes file:.(cpp)$ type:file repo:^github.com/foo/bar$@rev file:^file.cpp$',
```

Comment:
I think the dots in the `file:` query should be escaped with a backslash

Reply:
Ah, yup, and in general the string needs to be run through something like `escapeRegexChars()` (I made that up).